### PR TITLE
S3Sink example modified with cloudformation template

### PR DESCRIPTION
**Pull request by akshayar**
*Issue #, if available:*

*Description of changes:*
S3Sink example modified with CloudFormation template and additional instructions for local run. 
Removed the need to hardcode access and secret key. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by akshayar** [comment]
Typo errors as well in KinesisConnector and Windowing examples.

File: java/S3Sink/cloudformation/msf-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      RuntimeEnvironment: !Ref FlinkRuntimeEnvironment
+      ServiceExecutionRole: !GetAtt FlinkApplicationServiceExecutionRole.Arn
+      ApplicationConfiguration:
+        ApplicationCodeConfiguration:
+          CodeContent:
+            S3ContentLocation:
+              BucketARN: !Ref CodeBucketArn
+              FileKey: !Ref CodeKey
+          CodeContentType: 'ZIPFILE'
+        VpcConfigurations:
```
Comment (nicusX) [inline java/S3Sink/cloudformation/msf-deploy.yaml:48]: This example does not require VPC networking.

File: java/S3Sink/cloudformation/msf-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:DescribeLogGroups
+                  - logs:DescribeLogStreams
+                  - logs:PutLogEvents
+                Resource:  "*"
+        - PolicyName: VPCReadOnlyPermissions
```
Comment (nicusX) [inline java/S3Sink/cloudformation/msf-deploy.yaml:151]: (as above) VPC networking is not required

File: java/S3Sink/cloudformation/msf-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - ec2:DescribeVpcs
+                  - ec2:DescribeSubnets
+                  - ec2:DescribeSecurityGroups
+                  - ec2:DescribeDhcpOptions
+                Resource: "*"
+        - PolicyName: ENIReadWritePermissions
```
Comment (nicusX) [inline java/S3Sink/cloudformation/msf-deploy.yaml:162]: (as above) no VPC networking

File: java/S3Sink/README.md [hunk 4]
```diff
@@ -12,45 +12,149 @@ The Flink application consumes data in String format from a Kinesis Data Streams
...
+```
+ mvn clean compile exec:java  -Dexec.classpathScope="compile" \
+ -Dexec.mainClass="com.amazonaws.services.msf.StreamingJob" 
+
+```
+
+## Deploying using CloudFormation to Amazon Managed Service for Apache Flink
+### Pre-requisite
+1. Kinesis stream and S3 bucket. 
+2. Create subnets and security groups for the Flink application. If you are using private subnets , ensure that VPC endpoint for Kinesis and S3 is created.
```
Comment (nicusX) [inline java/S3Sink/README.md:109]: VPN networking is not required for this application

**Review by nicusX — CHANGES_REQUESTED** [review]
Brilliant work!
I suggest to remove VPC networking that is not required for an application that reads from KDS and writes to S3. Not all customers use VPC Endpoints, and it overcomplicates the setup for a simple example

File: java/S3Sink/cloudformation/msf-deploy.yaml (hunk 1 above)
Comment (akshayar) [reply java/S3Sink/cloudformation/msf-deploy.yaml]: Done.

File: java/S3Sink/cloudformation/msf-deploy.yaml (hunk 3 above)
Comment (akshayar) [reply java/S3Sink/cloudformation/msf-deploy.yaml]: Done and tested.

File: java/S3Sink/README.md (hunk 4 above)
Comment (akshayar) [reply java/S3Sink/README.md]: Done.

**Review by nicusX — APPROVED** [review]
LGTM

**Pull request merged**